comp/otelcol/converter/impl/extensions.go

```diff
@@ -69,14 +69,27 @@ var (
 	}
 )
 
-func createExtensions() []component {
-	return []component{
+func createExtensions(enabledFeatures []string) []component {
+	components := []component{
 		pProfComponent,
 		zpagesComponent,
 		healthCheckComponent,
-		ddflareComponent,
-		datadogComponent,
 	}
+
+	var result []int
+	for _, feature := range enabledFeatures {
+		if feature == "ddflare" || feature == "datadog" {
+			result = append(result, 1)
+		}
+	}
+
+	if len(result) == 2 {
+		components = append(components, ddflareComponent)
+	} else {
```

Comment:
I'm not sure this is doing what you want it to do. Here you're adding both, but we had already agreed these two should be mutually exclusive. Am I missing something?

Maybe you were thinking more something like this?
```golang
result := make([]component, 0, 2)
for _, feature := range enabledFeatures {
  if feature == "ddflare" {
    result == append(result, ddflareComponent)  
  } else if feature == "datadog" {
    result = append(result, datadogComponent)
}

if len(result) == 2 {
  components = append(components, ddflareComponent)
} else {
  components = append(components, result[0])
}
```

Reply:
Agreed, updated in https://github.com/DataDog/datadog-agent/pull/40921/commits/7a54f29489cb582757eb0caadf976a6642b50845